app/domain/authentication/authn_azure/application_identity.rb

```diff
@@ -0,0 +1,132 @@
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: ConstraintNotSupported, MissingAnnotationConstraint, IllegalConstraintCombinations
+
+    # This class defines an application identity of a given Conjur host.
+    # The constructor initializes an ApplicationIdentity object and validates that
+    # it is configured correctly.
+    # The difference between the validation in this constructor and
+    # the validation in ValidateApplicationIdentity is that here we validate that
+    # the application identity is configured correctly, and thus is a valid application
```

Comment:
It's good that you adding comments to clarify, but that this explanation is required is telling you that your names aren't clear enough yet.  Two different things should have different names. 

 It may be a sign that the logic isn't broken down in the best way too.

Also formatting nit: you should setup a hotkey in your editor so you can instantly square up things like this and keep clean margins.